plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -303,7 +324,11 @@ private List<YamlSource> getConfigFromSources(List<String> newSources) throws Co
     @Initializer(after = InitMilestone.SYSTEM_CONFIG_LOADED, before = InitMilestone.SYSTEM_CONFIG_ADAPTED)
     public static void init() throws Exception {
         detectVaultPluginMissing();
-        get().configure();
+        try {
+            get().configure();
+        } catch (ConfiguratorException e) {
+            throw new ConfigurationAsCodeBootFailure(e);
```

Comment:
This requires https://github.com/jenkinsci/jenkins/pull/8442 to work properly although it won't do any harm if not present